### ISSUE TITLE
clean up region_k_means.py

### ISSUE THREAD

**Issue by jGaboardi** [pysal/spopt#8]
[`region_k_means.py`](https://github.com/pysal/spopt/blob/master/spopt/region_k_means.py) includes several imports that are not used ([lines 2,3](https://github.com/pysal/spopt/blob/d45ec7d8c849df21b60e1d6b5674a08e6593d0d3/spopt/region_k_means.py#L2-L3)):
```python
from libpysal.weights.util import get_points_array_from_shapefile
from libpysal.io.fileio import FileIO as psopen
```

Also, defining `neighboring` within `is_neighbor()` ([line 66](https://github.com/pysal/spopt/blob/d45ec7d8c849df21b60e1d6b5674a08e6593d0d3/spopt/region_k_means.py#L66)) does not appear to be necessary:
```python
def is_neighbor(area, region, w):
    """Check if area is a neighbor of any member of region"""
    neighboring = False
    for member in region:
        if area in w[member]:
            return True
    return False
```

**Comment by sjsrey** [comment]
Great catches

**Comment by jGaboardi** [comment]
closed via #9
